### ISSUE TITLE
Escaping # in select/plural/etc

### ISSUE THREAD

**Issue by wopian** [ember-intl/ember-intl#1290]
- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

Before `5.0.0`, # did not need to be escaped in `select`. In `5.0.x` it now gets parsed the same as `plural`. 

In my usage, `#` was only ever used in `select` and hasn't been used in `plural` - where it would have the same correct behaviour seen in example `seven` in `4.x`

Escaping characters in 5.0.0 is now `'` instead of `\`, so `'#` should produce `#` (e.g syntax in examples four, five and ten)

### Environment

- Ember Version: 3.18.0
- Ember CLI Version: 3.18.0
- Ember Intl Version: 5.0.7
- Browser(s): N/A
- Node Version: 14.1.0

### Steps to Reproduce

1. `en-us.yaml`:

```
three: "Rank #{rank}"
four: "{type, select,
        anime {Rank '#{rank}}
        other {}}"
five: "{type, select,
        anime {Rank '#{rank}}
        manga {Rank '#{rank}}
        other {}}"
six: "{type, select,
        anime {Rank #{rank}}
        manga {Rank #{rank}}
        other {}}"
seven: "{num, plural,
        =0 {#{num} zero}
        =1 {#{num} one}
        other {#{num} other}}"
eight: "{num, plural,
        =0 {'#'{num} zero}
        =1 {'#'{num} one}
        other {'#'{num} other}}"
```

2. Template usage:

```html
<ul>
<li>3 {{t "three" rank=1}}</li>
<li>4 {{t "four" rank=1 type="anime"}}</li>
<li>5 {{t "five" rank=1 type="anime"}}</li>
<li>6 {{t "six" rank=1 type="anime"}}</li>
<li>7 {{t "seven" num=1}}</li>
<li>8 {{t "eight" num=1}}</li> 
</ul>
```

3. Output

```
3 Rank #1 (correct)
4 Rank {type, number}1
5 Rank {type, number}1
6 Rank NaN1 (Rank #1 in 4.x)
7 11 one (correct)
8 11 one (should be invalid? see nine)
```

4. The following in `en-us` produces an error:

Select `nine` uses incorrect syntax and should produce this error (but works in plural `eight` without error)
Plural `ten` uses correct syntax (but works in select `three`)

```yaml
nine: "{type, select,
        anime {Rank '#'{rank}}
        manga {Rank '#'{rank}}
        other {}}"
ten: "{num, plural,
        =0 {'#{num} zero}
        =1 {'#{num} one}
        other {'#{num}} other}"
```

[nine] An error occured (Expected "#", "'", "''", "\n", "{", "}", [^'], any character, argumentElement, double apostrophes, or tagElement but end of input found.) when extracting ICU arguments for '{type, select, anime {Rank '#'{rank} (Most Popular Anime)} manga {Rank '#'{rank} (Most Popular Anime)} other {}}'

[ten] An error occured (Expected "#", "'", "''", "\n", "{", "}", [^'], any character, argumentElement, double apostrophes, or tagElement but end of input found.) when extracting ICU arguments for '{num, plural, =0 {'#{num} zero} =1 {'#{num} one} other {'#{num} other}}'

**Comment by wopian** [comment]
Example repo https://github.com/wopian/ember-intl-test/tree/1290

**Comment by jasonmit** [comment]
@wopian unsure why the `intl-messageformat` is doing this but it appears you don't need to escape if you put a space after the `#`.  This may be spec valid, I'll need to look into it more.

```
{type, select,
        anime {Rank # {rank}}
        other {}}
```

FYI - it will not emit a space in between `#` and `rank`.

**Comment by wopian** [comment]
> FYI - it will not emit a space in between `#` and `rank`.

👍

**Comment by wopian** [comment]
Right now it emits a space between # and rank:

11 Rank NaN 1

```
"{type, select,
        anime {Rank # {rank}}
        manga {Rank # {rank}}
        other {}}"
```

12 Rank # 1

```
"Rank # {rank}"
```

**Comment by jasonmit** [comment]
Okay unsure what I was seeing then - sorry.  I'll need to open a bug upstream in intl-messageformat and figure out what's going on.

**Comment by jasonmit** [comment]
Figured it out, fix incoming.

**Comment by jasonmit** [comment]
Fixed in 5.0.9

**Comment by wopian** [comment]
Awesome! ❤️
